doc/introduction/circuits.rst

```diff
@@ -115,6 +117,40 @@ devices, the options are:
 
 For a plugin device, refer to the plugin documentation for available device options.
 
+**Defining custom wire labels**
+
+You can define your own wire labels instead of the default consecutive integers ``0, 1, ...``,
+and use them to address wires on the device. This is done by passing an iterable to the ``wires`` argument:
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit', wires=['wire1', 'wire2'], shots=1000, analytic=False)
+
+In the quantum function you can now use your own labels:
+
+.. code-block:: python
+
+    def my_quantum_function(x, y):
+        qml.RZ(x, wires='wire1')
+        qml.CNOT(wires=['wire1' ,'wire2'])
+        qml.RY(y, wires='wire2')
+        return qml.expval(qml.PauliZ('wire2'))
+
+Allowed wire labels can be of the following types:
+
+* *strings* like ``wires=['a', 'd', 'b', ...]`` or ``wires=['ancilla', 'q1', 'q2', ...]``,
+
+* *integers* like ``wires=[0, 4, 7]`` or ``wires=[-1, 0, 4]``
+
+* *floats* and other *numbers* like ``wires=[1., 2., 4.]``
+
+* *mixed types* like ``wires=['ancilla', -1, 0, 'q3']``
+
+.. note::
+
+    Some devices, such as hardware chips, may have a fixed number of wires,
+    which the length of the iterable of labels must match.
```

Comment:
Found this sentence a bit hard to read, although I don't have a good suggestion!